astro/src/content/docs/operate/roadmap/deprecation-policy.mdx

```diff
@@ -0,0 +1,57 @@
+---
+title: Deprecation Policy
+description: Learn more about the FusionAuth deprecation policy.
+navcategory: admin
+section: operate
+subcategory: roadmap
+---
+
+## Overview
+
+FusionAuth is an architectural component integrated into your applications or systems. Such integration means that when changes to FusionAuth occur, it can negatively affect your application. As developers ourselves, we strive to maintain backwards compatibility. 
+
+However, there are times when features or functionality must be removed. FusionAuth has [extensive release notes](/docs/release-notes/), which are the source of truth for changes to the FusionAuth system, and any deprecated features will be listed there.
+
+Beyond adding them to the release notes, as a measure of our commitment to backwards compatibility, FusionAuth commits to certain steps and time before removing.
+
+### What Is A Feature
+
+Here are examples of features that will follow this policy.
+
+* A documented API endpoint or parameter
+* A documented way to authenticate against an API
+* A documented themeable page
+
+### What Is Not A Feature
+
+Anything not documented in the [FusionAuth technical documentation](/docs) is not considered a feature and may be removed without following this policy.
+
+## Deprecation Steps 
+
+Before removing a feature, the FusionAuth team takes the following steps:
+
+* The feature is marked as deprecated in the documentation.
+* In the first release where the behavior is deprecated, the deprecation is noted in the release notes.
+* The documentation deprecation note remains for three full releases.
+* In the fourth full release, the feature and corresponding documentation are removed.
```

Comment:
Should we document this more of a minimum and not a hard number of releases? For example, we could ship 3 releases back to back - or 3 releases may be 3 months one year, and 5 months another. 

So perhaps we should state that a general rule is 3 releases, or 3 months. Or something like that. 

Should we document what a release is? Assuming a patch release does not apply here, so it would be a point release or major version release.

Reply:
Hmmm. So are you thinking whichever is a greater amount of time, 3 releases or 3 months? I'm not sure time matters very much because people can choose when they upgrade. So if there were four point releases in 2 weeks, we could deprecate a feature in the first release and remove it in the 4th, but people could choose not to upgrade.

Reply:
Could be either I guess... (3 months or 3 releases). But as a customer, I would assume the primary thing I want to know is how long do I have to plan and to react to something I read in the release notes. 

If everyone is ok with 3 releases, that is ok with me, just as long as we all agree that in theory that could be 2 months, or 6 months. 

If we think time is the primary thing a customer would need to prepare for a deprecation, then it feels like we should aim for a minimum amount of time which will generally equate to n releases. 

And the flip side is, if we end up not having a release for a few months because of unforeseen circumstances, I would not want to wait 3 releases if it has been 6 months as an example.

Reply:
I think we should be guaranteeing a minimum amount of calendar time that a customer has to do their internal dev work. So maybe we say something like
```
A deprecated feature will continue to exist for at least six months but may be removed at any time after that.
```

Reply:
Cool, cool. So the consensus is that the number of versions doesn't matter, it's how much time between when we say a feature is deprecated and when it is removed. Works for me. I'll rework the page. I think 3 months is an okay amount of time, because they can always wait longer (by not upgrading; the only cost will be if there's a security or functionality improvement). 

will follow up to confirm 3 months is acceptable to the eng org.

Reply:
Yeah, let's start with 3, aim for 6 and see if anyone yells really loud. 😎

Reply:
okay, all updated.